### DISCUSSION TITLE
Cythonize Mosaic submodule ?

### DISCUSSION THREAD

**Discussion by vincentsarago** [cogeotiff/rio-tiler#481] [Q&A]
The mosaic module iterate over numpy arrays and might be prone to some optimization if it was re-written in using Cython 🤷‍♂️ 

any though @kylebarron @geospatial-jeff

**Comment by kylebarron** [comment]
The module doesn't iterate over pixels, though, does it? It iterates over each array, filling any nodata areas of the result array? I'd need to look back at the code again, but assuming we're using vectorized numpy functions I doubt we'd see much of a performance gain.

**Reply by vincentsarago** [reply]
> It iterates over each array

☝️
